src/triage/cli.py

```diff
@@ -22,6 +23,65 @@ def __call__(self, args):
         print(CONFIG_VERSION)
 
 
+@Triage.register
+class Audition(Command):
+    """ Audition command to run or validate the config file
+    """
+    class Run(Command):
```

Comment:
Not sure but it seems like you might want to offer people a shortcut to `triage audition run`; you could give `Audition.__call__` the functionality of `Audition.Run.__call__`, such that `triage audition` is the same.

This is how `git stash` works, for which there's essentially an `argcmdr` recipe, in [[docs]](https://github.com/dssg/argcmdr#root-commands) and [[docs]](https://github.com/dssg/argcmdr#method-commands).

That said, it seems like users might like to always validate:

    class Audition(Command):

        def __init__(self, parser):
            parser.add_argument(
                '-c', '--config',
                metavar='path',
                type=argparse.FileType('r'),
                default='audition_config.yml',
                help="config file for audition",
            )
            parser.set_defaults(
                directory=None,
                validate=False,
            )

        @cachedproperty
        def runner(self):
            return AuditionRunner(
                yaml.load(self.args.config),
                self.args.directory,
            )

        def __call__(self, args):
            self['run'](args)

        @cmdmethod('-d', '--directory', help="...")
        @cmdmethod('-v', '--validate', action='store_true', help="validate first")
        def run(self, args):
            """..."""
            if args.validate:
                self['validate']()

            self.runner.run()

        @cmdmethod
        def validate(self):
            self.runner.validate()

In the above example, executing `triage audition` doesn't validate before running; but, it might make sense to make `validate=True` by default, instead:

    parser.set_defaults(
        …
        validate=True,
    )

    …

    @cmdmethod(…)
    @cmdmethod('--no-validate', action='store_false', dest='validate', help="...")
    def run(self, args):
        …

Reply:
I don't know if I would want `triage audition` to be a shortcut that does a potentially computationally-intensive task. My instinct is for that to tell me what I can do with the `triage audition` command.

Reply:
There's no one way to do it, so up to you guys.

However, it seems unconventionally verbose to make people type "run" after every command they actually want to run. The expectation of a command is that it does something; and, that you can learn about it, before running it even, by adding `-h` or `--help` to the end.

I don't know enough about audition to be able to predict what the possibilities are for the interface; and, of course, it can always change.

If there will be multiple `audition` actions, they should probably be given proper names, (rather than "run"). `validate` can either be one of these, or if it's something you might (always) want to perform in conjunction with your command action, then with a flag, as above.

But if it's really only ever going to be the one action, namely, "audition (my models)," then I think users would appreciate that we cut to the chase.

If they expected something else to happen, and regret how long it's taking to return, they can always interrupt it (ctrl+c).

(Potentially-destructive commands are another thing all together.)

Reply:
:+1: